prov/efa/src/efa_user_info.c

```diff
@@ -278,45 +278,82 @@ int efa_user_info_get_dgram(uint32_t version, const char *node, const char *serv
  */
 bool efa_user_info_should_support_hmem(int version)
 {
+	bool any_hmem, rdma_allowed;
+	char *extra_info = "";
+	int hmem_type;
 
 	/* Note that the default behavior of EFA provider is different between
-	 * libfabric API version.
+	 * libfabric API version when CUDA is used as HMEM system.
 	 *
 	 * For libfabric API version 1.17 and earlier, EFA provider does not
-	 * support FI_HMEM unless GPUDirect RDMA is available.
+	 * support FI_HMEM on CUDA unless GPUDirect RDMA is available.
 	 *
 	 * For libfabric API version 1.18 and later, EFA provider will claim support
-	 * of FI_HMEM as long as CUDA library is initialized and a CUDA device is
+	 * of FI_HMEM on CUDA as long as CUDA library is initialized and a CUDA device is
 	 * available. On an system without GPUDirect RDMA, the support of CUDA memory
 	 * is implemented by calling CUDA library. If user does not want EFA provider
 	 * to use CUDA library, the user can call use fi_setopt to set
 	 * FI_OPT_CUDA_API_PERMITTED to false.
 	 * On an system without GPUDirect RDMA, such a call would fail.
+	 *
+	 * For NEURON and SYNAPSEAI HMEM types, use_device_rdma is required no
+	 * matter the API version, as is P2P support.
 	 */
-	if (FI_VERSION_GE(version, FI_VERSION(1, 18)) && hmem_ops[FI_HMEM_CUDA].initialized) {
+	if (hmem_ops[FI_HMEM_CUDA].initialized && FI_VERSION_GE(version, FI_VERSION(1, 18))) {
 		EFA_INFO(FI_LOG_CORE,
-			"User is using API version >= 1.18. CUDA library and device are availble, claim support of FI_HMEM.");
-		return true;
+			"User is using API version >= 1.18. CUDA library and "
+			"device are available, claim support of FI_HMEM.");
+			/* For this API we can support HMEM regardless of
+			   use_device_rdma and P2P support, because we can use
+			   CUDA api calls.*/
+			return true;
+	} else if (hmem_ops[FI_HMEM_CUDA].initialized) {
```

Comment:
Nit. No need for `else` here because we already returned in previous branch

Reply:
removed